### PR TITLE
1169 resolved

### LINKED ISSUE #1169: Markers Windows not re-rendering content
I've got a Markers directive with a child Windows directive, and when I update the content in the model, it doesn't get reflected in any of the InfoWindows that are open (or closed for that matter).

Debugging the code led me to `WindowsParentModel.prototype.pieceMeal...` at line 5497, specifically `return _this.figureOutState...` at line 5517.  `figureOutState` is the method that determines whether the markers model has changed and returns an object `payload` containing marker adds, removals, and updates.  However, only adds and removals are handled when the promise returns
in lines 5520:
`return _async.each(payload.removals, function(child) {...`
and 5530:
`return _async.each(payload.adds, function(modelToAdd) {...`

Updates seem simply to be ignored and seem to be the reason for the lack of re-rendering.

I've hacked in a fix locally, but I'd prefer if someone more familiar with the code took a look and let me know if my analysis is correct.

I'm using angular-google-maps version 2.0.12.

### PR DESCRIPTION
**Pull request by nmccready**
#1169

### CODE REVIEW

**Comment by nmccready** [comment]
@zacronos @cthrax

File: src/coffee/directives/api/models/child/window-child-model.coffee [hunk 1]
```diff
@@ -30,10 +30,9 @@ angular.module('uiGmapgoogle-maps.directives.api.models.child')
           @scope.$on '$destroy', =>
             @destroy()
           $log.info @
-        #todo: watch model in here, and recreate / clean gObject on change
 
-        doShow: =>
-          if @scope.show == true
+        doShow: (wasOpen) =>
+          if @scope.show == true or wasOpen
```
Comment (cthrax) [inline src/coffee/directives/api/models/child/window-child-model.coffee]:
```coffeescript
if wasOpen
    @showWindow()
```

Is this correct? If it was open, don't you want to hide the window?

File: src/coffee/directives/api/models/child/window-child-model.coffee [hunk 2]
```diff
@@ -54,44 +58,46 @@ angular.module('uiGmapgoogle-maps.directives.api.models.child')
...
-                  _.delay => #appears to help animation chrome bug
-                    maybeMarker.setVisible false
-                    maybeMarker.setVisible @markerIsVisibleAfterWindowClose
-                  , 250
-              @gObject.close()
-              @model.show = false
-              if @scope.closeClick?
-                @scope.$evalAsync @scope.closeClick()
+
+          defaults = {}
```
Comment (nmccready) [inline src/coffee/directives/api/models/child/window-child-model.coffee]: basically moved stuff around so createGWin can play with an existing gObject and set its content without having to erase it.

File: src/coffee/directives/api/models/child/window-child-model.coffee (hunk 1 above)
Comment (nmccready) [inline src/coffee/directives/api/models/child/window-child-model.coffee]: No not always. In the case of this example the Window is moving and its content needs to be updated. However sometimes `@scope.show` can be undefined. So in the case that show is not defined then wasOpen should trump it.

**Comment by cthrax** [comment]
:+1:

**Comment by nmccready** [comment]
thumbs up mean merge?

**Comment by cthrax** [comment]
Means, it looks good to me. :) Unless you want to give @zacronos some time, I'm ok merging.

**Comment by nmccready** [comment]
This should be good ty

**Pull request merged**